### PR TITLE
Demo a possible modification of `funcCall`

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

Closes #<issue_number>.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [ ] I ensured that PR title is good enough for the changelog.
- [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [ ] I marked all done items in this checklist.

### CODE REVIEW

File: internal/util/observability/funccall.go [hunk 1]
```diff
@@ -19,29 +19,37 @@ import (
...
-//	    defer FuncCall(ctx)()
+//	    ctx, stop := FuncCall(ctx)
+//	    defer stop()
 //	    // ...
+//	}
 //
 // For the Go execution tracer, FuncCall creates a new region for the function call
 // and attaches it to the task in the context (or background task).
-func FuncCall(ctx context.Context) func() {
+func FuncCall(ctx context.Context, name string) (context.Context, func()) {
```
Comment (rumyantseva) [inline internal/util/observability/funccall.go:52]: @AlekSi PTAL

File: internal/util/observability/funccall.go [hunk 2]
```diff
@@ -65,5 +75,7 @@ func (fc *funcCall) leave() {
 		fc.region.End()
 	}
 
+	fc.span.End()
+
```
Comment (AlekSi) [inline internal/util/observability/funccall.go:79]: That's the part is what I'm currently thinking to do

File: internal/util/observability/funccall.go (hunk 1 above)
Comment (AlekSi) [inline internal/util/observability/funccall.go:52]: We already know the function name

**Review by AlekSi — COMMENTED** [review]
Give me a moment to update #4430

**Pull request closed without merging**